jekyll/_cci2/docker-hub-pull-through-mirror.adoc

```diff
@@ -20,60 +20,135 @@ Starting https://www.docker.com/blog/scaling-docker-to-serve-millions-more-devel
 
-2. *Reload Docker daemon* to apply the configuration.
+. *Obtain a TLS certificate for `your-mirror.example.com`* and *put the certificate and the private key into `/root/tls`* of the server.
 +
-`sudo systemctl restart docker.service`
+NOTE: This guide assumes you are obtaining a certificate issued by a well-known CA, not a self-signed certificate. Use of self-signed certificates has NOT been tested.
+
+. Run the command below to *start the registry*.
++
+[NOTE]
+====
+Replace `DOCKER_HUB_USERNAME` and `DOCKER_HUB_ACCESS_TOKEN` with the username for the Docker Hub account and the access token you obtained through https://hub.docker.com/settings/security, respectively.
+
+Replace `fullchain.pem` and `privkey.pem` with the filenames of your certificate and private key, respectively. *The certificate specified there must be correctly chained enough for tracing the certification path from the leaf to the root*.
+====
++
+[source,bash]
+----
+sudo docker run \
+  -d \
+  -p 443:5000 \
+  --restart=always \
+  --name=through-cache-secure \
+  -v /root/tls:/data/tls \
+  -e REGISTRY_PROXY_REMOTEURL="https://registry-1.docker.io" \
+  -e REGISTRY_PROXY_USERNAME=DOCKER_HUB_USERNAME \
+  -e REGISTRY_PROXY_PASSWORD=DOCKER_HUB_ACCESS_TOKEN \
+  -e REGISTRY_HTTP_TLS_CERTIFICATE=/data/tls/fullchain.pem \
+  -e REGISTRY_HTTP_TLS_KEY=/data/tls/privkey.pem \
+  registry
+----
+
+. Finally, *make sure that the TCP port 443 (i.e. HTTPS) is open and accessible*. For better security, we recommend you to open the port only to Nomad clients and VMs for `machine` executors and remote docker engines.
++
+On AWS, for example, you will need to make sure that both Security Groups and the firewall at the OS level are configured correctly to accept connections from Nomad clients and VMs for `machine`/`setup_remote_docker` jobs over HTTPS.
+
+==== Plan for renewal of TLS certificates
+
+You will need to renew TLS certificates periodically. These are the steps required to renew certificates.
+
+. Update the certificate and the private key in `/root/tls`.
+
+. Run `docker restart through-cache-secure`.
 
-Now you should be able to see that Docker images from Docker Hub are downloaded through the cache server. As a side effect, you should be able to see that private images for the user, configured for the cache server can be downloaded without explicit authentications.
+Technically, this can be automated. For example, if you are using Let's Encrypt for your certificates, you can setup a cron task that executes `certbot renew` and the steps above.
 
-== Configure VM Service to let Machine/Remote Docker VMs use the Pull Through Cache Server
+== Configure Nomad clients to use the pull through cache registry (run this for _each_ Nomad client)
```

Comment:
```suggestion
== Configure Nomad Clients to use the Pull Through Cache Registry (run for _each_ Nomad client)
```

Reply:
Sorry! Just a bunch of changes to fix titles to use title case then good to go from my point of view :-D